dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/SourcesToStartingPoints.scala

```diff
@@ -9,48 +9,132 @@ import io.shiftleft.semanticcpg.language.operatorextension.allAssignmentTypes
 import io.shiftleft.semanticcpg.utils.MemberAccess.isFieldAccess
 import org.slf4j.LoggerFactory
 
-import java.util.concurrent.{ForkJoinPool, ForkJoinTask, RecursiveTask, RejectedExecutionException}
+import java.util.concurrent.*
+import scala.collection.mutable.ListBuffer
 import scala.util.{Failure, Success, Try}
 
 case class StartingPointWithSource(startingPoint: CfgNode, source: StoredNode)
-
+case class UsageInput(src: StoredNode, typeDecl: TypeDecl, astNode: AstNode)
+case class ResultSummary(result: List[StartingPointWithSource], methodTasks: List[UsageInput])
 object SourcesToStartingPoints {
 
   private val log = LoggerFactory.getLogger(SourcesToStartingPoints.getClass)
 
   def sourceTravsToStartingPoints[NodeType](sourceTravs: IterableOnce[NodeType]*): List[StartingPointWithSource] = {
-    val fjp = ForkJoinPool.commonPool()
+    val executorService = Executors.newWorkStealingPool()
+    val completionService =
+      new ExecutorCompletionService[Unit](executorService)
     try {
-      fjp.invoke(new SourceTravsToStartingPointsTask(sourceTravs: _*)).distinct
+      val sources: List[StoredNode] = sourceTravs
+        .flatMap(_.iterator.toList)
+        .collect { case n: StoredNode => n }
+        .dedup
+        .toList
+        .sortBy(_.id)
+      val firstResultAggregator       = SourceStartingPointResultAggregator(sources.size)
```

Comment:
I really don't know the entire logic that is getting processed in the flow. Hence the names are first and second. I moved out the part in which all the tasks were taking time looking at the profiler thread dump. Refer to this [section](https://github.com/joernio/joern/pull/4429/files#diff-b887bf35647ceef08727f3fa1753c2cdcb473e966b53d598bbdf1b2f186dd6d4L131-L150). So instead of processing them in the same task. I created separate chunks of tasks for these sections and executed those tasks in the second run by taking individual `Method` nodes per task.